src/platform/lumin-runtime/utilities/resource-download.js

```diff
@@ -0,0 +1,73 @@
+// The purpose of the resource downloader is to download image or model from remote location (url)
+
+import { ui, INVALID_RESOURCE_ID } from 'lumin';
+
+import executor from './executor.js';
+
+import { logError } from '../../../util/logger.js';
+import saveResource from '../../../util/download.js';
+
+const DOWNLOAD_COLOR_MASK = [0.1, 0.1, 0.1, 0.1];
+const DEFAULT_COLOR_MASK = [1, 1, 1, 1];
+
+function createSpinner (prism) {
+    const spinner = executor.createNode(ui.UiLoadingSpinner, 'Create', prism, ui.LoadingSpinnerType.k2dSpriteAnimation);
+    const [w, h] = executor.callNativeFunction(spinner, 'getSize');
+    const [x, y, z] = executor.callNativeFunction(spinner, 'getLocalPosition');
+
+    executor.callNativeAction(spinner, 'setLocalPosition', [x - (w / 2), y - (h / 2), z]);
+    return spinner;
+}
+
+function addMaskAndSpinner (element, prism) {
+    // Set color mask
+    executor.callNativeAction(element, 'setColor', DOWNLOAD_COLOR_MASK);
+
+    // Add downloading spinner
+    const spinner = createSpinner(prism);
+    executor.callNativeAction(element, 'addChild', spinner);
+
+    return spinner;
+}
+
+function removeMaskAndSpinner (element, prism, spinner, properties) {
+    // Remove color mask
+    const color = properties.color ? properties.color : DEFAULT_COLOR_MASK
```

Comment:
This will override any color that the developer has set on this element. We should probably save off the current color first and then restore it here to be safe

Reply:
That is what the code do. If property `color` has value we use the value if not we use the default.

Reply:
Oops...I didn't look closely enough. I didn't see the old value being saved off but that's because we can get it from the props.